library/Zend/XmlRpc/Value.php

```diff
@@ -486,13 +486,17 @@ protected static function _createSimpleXMLElement(&$xml)
      */
     protected static function _extractTypeAndValue(SimpleXMLElement $xml, &$type, &$value)
     {
-        list($type, $value) = each($xml);
+        $type = key($xml);
+        $value = current($xml);
+        next($xml);
 
-        if (!$type and $value === null) {
+        if ($type === null && $value === false) {
```

Comment:
These are the two values that would be set if `key` and `current` above were at the end of the array (or the array was empty).